### ISSUE TITLE
TestTTLFinalizer failure

### ISSUE THREAD

**Issue by kyessenov** [istio/istio#4304]
`pkg/cache` test suite is flaky. Please fix the flakiness, or we have to disable the test:
```
I0315 21:33:26.971] === RUN   TestLRUEvicter
I0315 21:33:26.972] --- FAIL: TestLRUEvicter (0.01s)
I0315 21:33:26.972] 	cache_test.go:255: Got entry, expecting it to have been evicted
I0315 21:33:26.975] === RUN   TestLRUEvictExpired
I0315 21:33:26.975] --- PASS: TestLRUEvictExpired (0.05s)
I0315 21:33:26.976] === RUN   TestLRUFinalizer
I0315 21:33:26.976] --- PASS: TestLRUFinalizer (0.00s)
I0315 21:33:26.976] === RUN   TestLRUBehavior
... skipping 57 lines ...
I0315 21:33:27.005] === RUN   TestTTLEvicter
I0315 21:33:27.006] --- PASS: TestTTLEvicter (0.01s)
I0315 21:33:27.006] === RUN   TestTTLEvictExpired
I0315 21:33:27.006] --- PASS: TestTTLEvictExpired (0.05s)
I0315 21:33:27.006] === RUN   TestTTLFinalizer
I0315 21:33:27.008] --- PASS: TestTTLFinalizer (0.00s)
I0315 21:33:27.008] FAIL
I0315 21:33:27.008] FAIL	istio.io/istio/pkg/cache	0.689s
```
https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4300/istio-unit-tests/2408/

**Comment by kyessenov** [comment]
Happened again in @mandarjog 's PR ...

**Comment by geeknoid** [comment]
@ZackButcher Can we consider this fixed?

**Comment by kyessenov** [comment]
Please link the PR that fixes the tests. For now, the tests are disabled (can be moved to post-submit, if desired).

**Comment by geeknoid** [comment]
So Kuat, you disabled all the cache tests. Did they all show flakyness?

**Comment by kyessenov** [comment]
3 tests in this package flaked in unrelated PRs. I looked in the code, and they appear to make assumptions about timing and uninterrupted execution (not true in our test environment). I then looked at other tests that use the same pattern, and marked them as skipped. It takes quite a bit of work to discover why and when tests flake, so it's better to apply the same fix across all tests in this package.
